### PR TITLE
Fixes #22561 - consolidate repo create/update apipie docs

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by ehelms** [comment]
[test katello]

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
[test katello]

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 1]
```diff
@@ -254,23 +258,7 @@ def export
     end
 
     api :PUT, "/repositories/:id", N_("Update a repository")
-    param :name, String, :desc => N_("New name for the repository")
-    param :id, :number, :required => true, :desc => N_("repository ID")
```
Comment (jturel) [inline app/controllers/katello/api/v2/repositories_controller.rb]: This should work fine once the `id` param is added to the :repo param group :)

Comment (jturel) [reply app/controllers/katello/api/v2/repositories_controller.rb]: [vagrant@centos7-hammer-devel ~]$ hammer repository update --id=2 --name=Zoo1
Could not update the repository:
  Error: Unrecognised option '--id'.

Comment (jlsherrill) [reply app/controllers/katello/api/v2/repositories_controller.rb]: doh, good catch!

**Comment by jlsherrill** [comment]
updated.

**Comment by jturel** [comment]
Update works now:
```
vagrant@centos7-hammer-devel ~]$ hammer repository update --id=2 --name=ZooZoo --product="Test Product" --organization-id=1
Repository updated
```

Expected param is there:
```
[vagrant@centos7-hammer-devel ~]$ hammer repository update --help | grep ignore-global
 --ignore-global-proxy IGNORE_GLOBAL_PROXY                 If true, will ignore the globally configured proxy when syncing.
```

```
[vagrant@centos7-hammer-devel ~]$ hammer repository update --id=2 --ignore-global-proxy=true --product="Test Product" --organization-id=1
Could not update the repository:
  Missing arguments for 'name'
[vagrant@centos7-hammer-devel ~]$ hammer repository update --id=2 --name="ZoOZoo" --ignore-global-proxy=true --product="Test Product" --organization-id=1
Repository updated
```

This change made 'name' required on update - I don't think that's intended, right @jlsherrill  ?

**Comment by jlsherrill** [comment]
@jturel good catch!  Updated to pull name out of the 'repo' param group.

**Comment by jturel** [comment]
```
[vagrant@centos7-hammer-devel ~]$ hammer repository create --name='TestRepo' --product-id=1 --organization-id=1 --content-type=yum
Repository created
[vagrant@centos7-hammer-devel ~]$ hammer repository update --id=2 --ignore-global-proxy=true --product="Test Product" --organization-id=1
Repository updated
```

looks good to me. APJ

**Comment by jturel** [comment]
[test katello]

**Pull request merged**